rust/vectordb/src/table.rs

```diff
@@ -470,44 +482,35 @@ impl NativeTable {
         version: u64,
         write_store_wrapper: Option<Arc<dyn WrappingObjectStore>>,
         params: ReadParams,
+        read_consistency_interval: Option<std::time::Duration>,
     ) -> Result<Self> {
         // patch the params if we have a write store wrapper
         let params = match write_store_wrapper.clone() {
             Some(wrapper) => params.patch_with_store_wrapper(wrapper)?,
             None => params,
         };
-        let dataset = Dataset::checkout_with_params(uri, version, &params)
-            .await
-            .map_err(|e| match e {
-                lance::Error::DatasetNotFound { .. } => Error::TableNotFound {
-                    name: name.to_string(),
-                },
-                e => Error::Lance {
-                    message: e.to_string(),
-                },
-            })?;
+        let dataset = DatasetBuilder::from_uri(uri)
+            .with_version(version)
+            .with_read_params(params)
+            .load()
+            .await?;
+        let dataset = DatasetConsistencyWrapper::new_time_travel(dataset, version);
+
         Ok(Self {
             name: name.to_string(),
             uri: uri.to_string(),
-            dataset: Arc::new(Mutex::new(dataset)),
+            dataset,
             store_wrapper: write_store_wrapper,
+            read_consistency_interval,
         })
     }
 
     pub async fn checkout_latest(&self) -> Result<Self> {
-        let dataset = self.clone_inner_dataset();
-        let latest_version_id = dataset.latest_version_id().await?;
-        let dataset = if latest_version_id == dataset.version().version {
-            dataset
-        } else {
-            dataset.checkout_version(latest_version_id).await?
-        };
-
+        let mut dataset = self.dataset.duplicate().await;
+        dataset.as_latest(self.read_consistency_interval).await?;
```

Comment:
Maybe, instead of reusing `self.read_consistency_interval`, we could have this parameter accept `read_consistency_interval` as an argument?

I think users should generally only call `checkout_latest` if `self.read_consistency_interval` is `None` anyways.

As a perk, then we don't have to store it on this object.

Reply:
There are two uses of `checkout_latest`:
1. To force an update check if `self.read_consistency_interval` is `None`
2. To switch from "time travel mode" into "latest mode".

The latter could happen when `read_consistency_interval` was not `None`. I like keeping it out of the parameters here because it is meant to be a connection-level parameter.